src/ws2st/configurations/wone_rules.cpp

```diff
@@ -0,0 +1,186 @@
+            WashRefactoringAction(&forces::AddForceVectorMatcher, forces::HandleRegisterForcesVector),
+            WashRefactoringAction(&forces::AddForceScalarMatcher, forces::HandleRegisterForcesScalar),
+            
+            // Register Variables with the simulation
+            WashRefactoringAction(&variables::RegisterVariableMatcher, &variables::HandleRegisterVariable),
+            WashRefactoringAction(&variables::RegisterVariableNoInitMatcher, &variables::HandleRegisterVariable),
+        },
+        // 4th pass: registration, gets
+        {   
+            &AllFiles,
+            // Rewrite IO functions which inspect the list of forces/force names
+            WashRefactoringAction(&meta::DefineForceAccessFnMatcher, &meta::DefineForceAccessFnsWithCornerstone),
+            // Calls to get a force
+            WashRefactoringAction(&forces::GetForceScalarMatcher, forces::HandleGetForceScalar),
+            WashRefactoringAction(&forces::GetForceVectorMatcher, forces::HandleGetForceVectorWithCornerstone),
+            // Calls to get pre-defined particle properties
+            WashRefactoringAction(&forces::GetPosMatcher, forces::HandleGetPosWithCornerstone),
+            WashRefactoringAction(&forces::GetVelMatcher, forces::HandleGetVelWithCornerstone),
+            WashRefactoringAction(&forces::GetAccMatcher, forces::HandleGetAccWithCornerstone),
+        
+            WashRefactoringAction(&forces::GetDensityMatcher, forces::HandleGetDensity),
+            WashRefactoringAction(&forces::GetMassMatcher, forces::HandleGetMass),
+            WashRefactoringAction(&forces::GetSmoothingLengthMatcher, forces::HandleGetSmoothingLength),
+            WashRefactoringAction(&forces::GetIdMatcher, forces::HandleGetId),
+
+            // Calls to get a variable
+            WashRefactoringAction(&variables::GetVariableMatcher, &variables::HandleGetVariable),
+            WashRefactoringAction(&variables::GetVariableRefMatcher, &variables::HandleGetVariableRef),
+            // Replacement to add the force definitions
+            WashRefactoringAction(&forces::InsertForcesDefinitionMatcher, &forces::HandleInsertForcesDefinitionWithCornerstone),
+            WashRefactoringAction(&variables::InsertVariablesDeclarationMatcher, &variables::HandleInsertVariablesDeclaration),
+
+            WashRefactoringAction(&meta::SimulationVecTMatcher, &meta::HandleSimulationVecTMatcher),
+
+            // Cornerstone Specific stuff - just needs the forces found as prereq
+            WashRefactoringAction(&cornerstone::DataSetupDecl, &cornerstone::HandleSetupDataWithCornerstone),
+            WashRefactoringAction(&cornerstone::ParticleRecalculateNeighbours, &cornerstone::HandleRecalculateNeighboursWithCornerstone),
+            WashRefactoringAction(&cornerstone::SyncDomainCall, &cornerstone::HandleSyncDomainWithCornerstone),
+            WashRefactoringAction(&cornerstone::ExchangeAllHalos, &cornerstone::HandleExchangeAllHalosWithCornerstone),
```

Comment:
What's the difference between this and `InsertInitialHaloExchange` from 6th pass?

Reply:
This is a holdover from when we just exchanged all force halos between kernels. Not needed anymore, and shouldn't have any effect on running. Will remove.